Forge/src/main/java/me/ferdz/placeableitems/PlaceableItems.java

```diff
@@ -15,19 +15,31 @@
 import org.apache.logging.log4j.Logger;
 
 @Mod(PlaceableItems.MODID)
-public class PlaceableItems {
+public final class PlaceableItems {
 
     public static final String MODID = "placeableitems";
     private static final boolean GENERATE_WIKI = "true".equals(System.getenv().get("GENERATE_WIKI"));
 
     // Directly reference a log4j logger.
-    private static final Logger LOGGER = LogManager.getLogger();
+    public static final Logger LOGGER = LogManager.getLogger();
+
+    private static PlaceableItems instance;
+
+    private final ItemPlaceHandler placeHandler;
 
     public PlaceableItems() {
+        instance = this;
+
         // Register ourselves for server and other game events we are interested in
         FMLJavaModLoadingContext.get().getModEventBus().addListener(this::onCommonSetup);
-        DistExecutor.runWhenOn(Dist.CLIENT, () -> () -> FMLJavaModLoadingContext.get().getModEventBus().register(ClientListener.get()));
-        MinecraftForge.EVENT_BUS.register(new ItemPlaceHandler());
+        MinecraftForge.EVENT_BUS.register(this.placeHandler = new ItemPlaceHandler());
+
+        DistExecutor.runWhenOn(Dist.CLIENT, () -> () -> {
+            ClientListener clientListener = ClientListener.get();
+
+            FMLJavaModLoadingContext.get().getModEventBus().register(clientListener.getModBus());
+            MinecraftForge.EVENT_BUS.register(clientListener.getForgeBus());
```

Comment:
This is starting to feel a bit messy. I would suggest moving the registering responsibility inside of ClientListener. Such as when .get() is called, the event buses are correctly registered without this constructor having to do it manually. Also this would allow you to get rid of the `getModBus` & `getForgeBus`

Reply:
I suppose that would work. It would also remove the need to have `@SubscribeEvent` annotations and public static methods.